Scenes/Enemies/TemplateEnemy/TemplateEnemy.gd

```diff
@@ -26,22 +29,28 @@ func set_debug(val: bool):
 func _physics_process(delta):
 	if goal.distance_to(global_position) < ARRIVAL_BUFFER:
 		handle_goal_arrival()
-	if not path:
-		update_path()
+#	if not path:
+#		update_path(get_world_2d().get_navigation_map())
 	if path.size() > 0:
 		if (path[next_destination_idx].distance_to(global_position)) < ARRIVAL_BUFFER:
 			if path.size() > next_destination_idx + 1:
 				next_destination_idx += 1
 
 		var direction = (path[next_destination_idx] - global_position).normalized()
 		var movement = direction * speed
-		$Debug/MovementLine2d.points = PoolVector2Array([Vector2(0,0), direction * 100])
-		move_and_collide(movement * delta)
+		
+		move_and_slide(movement)
+		if debug_mode:
+			$Debug/MovementLine2d.points = PoolVector2Array([Vector2(0,0), direction * 100])
+			$Debug/PathLine2d.points = path
+	else:
```

Comment:
I assume this is for debug purposes to see if the path was empty, so we could either get rid of it or change to `print_debug()`?

Reply:
Yes it was that and I forgot to remove, good catch